benchs/data-parser/benches/deserialize.rs

```diff
@@ -13,63 +16,49 @@ fn bench_deserialize(c: &mut Criterion) {
 
     // Benchmarks for different serialization methods
     let parametric_matrix = SerializationType::iter()
-        .map(|ser_type| (ser_type, CompressionType::None, Level::Default))
+        .map(|serialization_type| {
+            (serialization_type, DEFAULT_COMPRESSION_STRATEGY.clone())
+        })
         .collect::<Vec<_>>();
 
     // Pre-serialize data for each serialization type
-    let serialized_data: Vec<(
-        SerializationType,
-        CompressionType,
-        Level,
-        Vec<u8>,
-    )> = parametric_matrix
+    let serialized_data: Vec<_> = parametric_matrix
         .iter()
-        .map(
-            |(serialization_type, compression_type, compression_level)| {
-                let test_block = generate_test_block();
-                let data_parser = DataParserBuilder::new()
-                    .with_compression(*compression_type)
-                    .with_compression_level(*compression_level)
-                    .with_serialization(*serialization_type)
-                    .build();
+        .map(|(serialization_type, compression_strategy)| {
```

Comment:
`compression_strategy` is sort of a replacement for `compression_type`. 

Also, I removed the `compression_level` for simplicity since we care about the`Fastest` level for the most part. If we are keen on supporting all combinations, we need to define statically them with a suffix like `ZLibFastestCompressionStrategy`, `ZLibBestCompressionStrategy`, et al behind the `bench-helpers` feature flag of course. This might not be ergonomic but it will allow us drastically save a tonne of byte code for our main code.